stylelint-config-dcos/README.md

```diff
@@ -1,76 +1,279 @@
 
-## Example
-```LESS
-& when (@status-bar-enabled) {
+## Table of Contents
 
-  // A single newline is required before beginning a nested declaration.
-  @keyframes candy-stripe {
+  1. [Terminology](#terminology)
+    - [Rule Declaration](#rule-declaration)
+    - [Selectors](#selectors)
+    - [Properties](#properties)
+  1. [Rules](#rules)
+    - [Directory Structure](#directory-structure)
+    - [Formatting](#formatting)
+    - [Comments](#comments)
+    - [Writing Useful Class Selectors](#writing-useful-class-selectors)
+    - [Ordering](#ordering)
+      - [Property declarations](#property-declarations)
+      - [Pseudo-classes and pseudo-elements](#pseudo-classes-and-pseudo-elements)
+      - [Nested selectors](#nested-selectors)
+    - [Variables](#variables)
+    - [Mixins](#mixins)
+    - [Extend directive](#extend-directive)
 
-    0% {
-      // Units on zero-values are prohibited.
-      background-position: 0 0;
-    }
+## Terminology
 
-    100% {
-      background-position: 16px 0;
-    }
+### Rule declaration
+
+A “rule declaration” is the name given to a selector (or a group of selectors) with an accompanying group of properties. Here's an example:
+
+```css
+.listing {
+  font-size: 18px;
+  line-height: 1.2;
+}
+```
+
+### Selectors
+
+In a rule declaration, “selectors” are the bits that determine which elements in the DOM tree will be styled by the defined properties. Selectors can match HTML elements, as well as an element's class, ID, or any of its attributes. Here are some examples of selectors:
+
+```css
+.my-element-class {
+  /* ... */
+}
+
+[aria-hidden] {
+  /* ... */
+}
+```
+
+### Properties
+
+Finally, properties are what give the selected elements of a rule declaration their style. Properties are key-value pairs, and a rule declaration can contain one or more property declarations. Property declarations look like this:
+
+```css
+/* some selector */ {
+  background: #f1f1f1;
+  color: #333;
+}
+```
+
+## Rules
+
+### Directory Structure
+
+Stylesheets should follow the current organization where there is a `styles.less` and `variables.less` inside a parent directory which.
+
+### Formatting
+
+* Use soft tabs (2 spaces) for indentation
+* Alphabetize properties in declarations
+* Prefer dashes over camelCasing in class names.
+* Do not use ID selectors
+* When using multiple selectors in a rule declaration, give each selector its own line.
+* Put a space before the opening brace `{` in rule declarations
+* In properties, put a space after, but not before, the `:` character.
+* Put closing braces `}` of rule declarations on a new line
+* Put blank lines between rule declarations
+
+**Bad**
+```less
+.avatar{
+    border-radius:50%;
+    border:2px solid white; }
+.no, .nope, .not_good {
+    // ...
+}
+#lol-no {
+  // ...
+}
+```
+
+**Good**
+```less
+.avatar {
+  border: 2px solid #fff;
+  border-radius: 50%;
+}
+
+.one,
+.selector,
+.per-line {
+  // ...
+}
+```
+
+### Comments
```

Comment:
These are really cool! It would be especially awesome to have this documentation in our DC/OS UI pattern library/styleguide thing.

How do you propose we roll this out? It would take some time to document all of our LESS components, so I think we would need to set aside time to do this.